aspnetcore/blazor/hybrid/class-libraries.md

```diff
@@ -0,0 +1,514 @@
+---
+title: Share assets across web and native clients using a Razor class library (RCL)
+author: guardrex
+description: Learn how to share Razor components, C# code, and static assets across web and native clients using a Razor class library (RCL).
+monikerRange: '>= aspnetcore-6.0'
+ms.author: riande
+ms.custom: mvc
+ms.date: 07/07/2022
+uid: blazor/hybrid/class-libraries
+---
+# Share assets across web and native clients using a Razor class library (RCL)
+
+Use a Razor class library (RCL) to share Razor components, C# code, and static assets across web and native client projects.
+
+This article builds on the general concepts found in the following articles:
+
+* <xref:blazor/components/class-libraries>
+* <xref:razor-pages/ui-class>
+
+The examples in this article share assets between a Blazor Server app and a .NET MAUI Blazor app in the same solution:
+
+* Although a Blazor Server app is used in examples and described by the text as the "web" client, the guidance applies equally to Blazor WebAssembly apps.
+* The example projects are in the same solution, but an RCL can supply shared assets to projects outside of the same solution.
+* The RCL example is added as a project to the solution, but any RCL can be published as a NuGet package. A NuGet package can supply shared assets to web and native client projects.
+* The order that the projects are created isn't important. However, projects that rely on an RCL for assets must create a project reference to the RCL *after* the RCL is created.
+* The following project names are used for the projects in this article, which become part of namespace and assembly references in the examples:
+  * RCL: `SharedLibrary`
+  * Blazor Hybrid (.NET MAUI Blazor): `BlazorNative`
+  * Blazor Server: `BlazorWeb`
+
+For guidance on creating an RCL, see <xref:blazor/components/class-libraries>. Optionally, access the additional guidance on RCLs that apply broadly to ASP.NET Core apps in <xref:razor-pages/ui-class>.
+
+## Advanced sample app
+
+For an advanced example of the scenarios described in this article, see the .NET Podcasts sample app:
+
+* [GitHub repository (`microsoft/dotnet-podcasts`)](https://github.com/microsoft/dotnet-podcasts)
+* [Running sample app (Azure Container Apps Service)](https://dotnetpodcasts.azurewebsites.net/)
+
+The .NET Podcasts app showcases the following technologies:
+
+* [.NET](https://dotnet.microsoft.com/download/dotnet)
+* [ASP.NET Core](https://dotnet.microsoft.com/apps/aspnet)
+* [Blazor](https://dotnet.microsoft.com/apps/aspnet/web-apps/blazor)
+* [.NET MAUI](https://dotnet.microsoft.com/apps/maui)
+* [Azure Container Apps](https://azure.microsoft.com/services/container-apps/)
+* [Orleans](/dotnet/orleans/overview)
+
+## Share web UI Razor components, code, and static assets
```

Comment:
If we're cutting so much of the step-by-step, I'm concerned about not placing the sample app at `dotnet/blazor-samples`. If a future release has a project template for this, we'd merely drop the sample from that release version in the samples repo and from the doc. Another thing is that if .NET MAUI with Blazor Server is going to be the most common use case, the sample app could be quite speedy 🏃 for devs as a starting point. Otherwise, it takes like ... mmmm 🤔 .... it could take an hour for someone to build from scratch.